services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -1520,6 +1522,61 @@ public void migrateStore(String srcClusterName, String destClusterName, String s
     versionMigrationConsumer.accept(storeName);
   }
 
+  /**
+   * Clear KILL messages from a participant system store.
+   */
+  public void clearIngestionKillMessageAndVerify(String clusterName, String versionTopicName) {
+    long startTs = System.currentTimeMillis();
+    ParticipantMessageKey key = new ParticipantMessageKey();
+    key.messageType = ParticipantMessageType.KILL_PUSH_JOB.getValue();
+    key.resourceName = versionTopicName;
+    ParticipantMessageValue value = RetryUtils.executeWithMaxAttemptAndExponentialBackoff(
+        () -> participantStoreClientsManager.getReader(clusterName).get(key).get(),
+        3,
+        Duration.ofMillis(100),
+        Duration.ofSeconds(1),
+        Duration.ofSeconds(15),
+        Collections.singletonList(Exception.class));
+    if (value == null) {
+      return;
+    }
+
+    KillPushJob killPushJobMessage = (KillPushJob) value.messageUnion;
+    LOGGER.info(
+        "Deleting kill message for store-version: {} in cluster: {}. Message age: {}",
+        versionTopicName,
+        clusterName,
+        System.currentTimeMillis() - killPushJobMessage.getTimestamp());
+
+    // Use RetryUtils to add timed-wait
+    RetryUtils.executeWithMaxAttemptAndExponentialBackoff(
+        () -> deleteParticipantStoreKillMessage(clusterName, versionTopicName),
+        1,
+        Duration.ofSeconds(1),
+        Duration.ofSeconds(3),
+        Duration.ofSeconds(30),
+        Collections.singletonList(Exception.class));
+
+    // wait for kill message to be removed
+    RetryUtils.executeWithMaxAttemptAndExponentialBackoff(() -> {
```

Comment:
I understand that the first `RetryUtils. executeWithMaxAttemptAndExponentialBackoff` is useful to reduce the likelihood of race where we try to migrate store again when the previous attempt just failed (kill is written but still on its way to be reflected). However, I think we can remove `executeWithMaxAttemptAndExponentialBackoff` or reduce the attempt & duration for verification of the delete and just rely on admin channel retry.

Reply:
Since we've MT admin channel and assuming this is not frequent event, should 1m45sec be a concern?

Reply:
But why wait for up to a minute here when we know we can give up the thread to process other admin messages and come back and check again later. If we are worried that an idle admin channel could end up spamming the topic with lots delete kill messages then we can still keep this but reduce the attempt and duration.

Reply:
I believe we should give reasonable time to finish the op. We can certainly reduce the total time.